omni-relayer/src/startup/evm.rs

```diff
@@ -110,19 +119,50 @@ pub async fn start_indexer(
         chain_kind
     );
 
-    let mut stream = ws_provider.subscribe_logs(&filter).await?.into_stream();
-    while let Some(log) = stream.next().await {
-        process_log(
-            chain_kind,
-            &mut redis_connection,
-            &http_provider,
-            log,
-            expected_finalization_time,
-        )
-        .await;
-    }
+    loop {
+        let ws_provider = match ProviderBuilder::new()
+            .on_ws(WsConnect::new(&rpc_ws_url))
+            .await
+        {
+            Ok(provider) => provider,
+            Err(err) => {
+                error!(
+                    "{chain_kind:?} WebSocket connection failed: {}, retrying...",
+                    err
+                );
+                tokio::time::sleep(tokio::time::Duration::from_secs(5)).await;
```

Comment:
This can be refactored using macros.
See this example 
https://github.com/Near-One/eth2-on-near-client-validator/blob/48fe316da190573f523e37700c35ad557b38db67/src/eth2_validator.rs#L15